snapcraft/tests/unit/test_lxd.py

```diff
@@ -537,6 +537,49 @@ def make_containerbuild(self):
                            project_options=self.project_options,
                            remote=self.remote)
 
+    @patch('snapcraft.internal.lxd.Containerbuild._container_run')
+    @patch('snapcraft.internal.common.is_snap')
+    def test_inject_snap_existing_container(self,
+                                            mock_is_snap,
+                                            mock_container_run):
+        mock_is_snap.side_effect = lambda: True
+        mock_container_run.side_effect = lambda cmd, **kwargs: cmd
+
+        fake_snapd = fixture_setup.FakeSnapd()
+        self.useFixture(fake_snapd)
+        fake_snapd.snaps_result = [
+            {'name': 'core',
+             'confinement': 'strict',
+             'id': '2kkitQurgOkL3foImG4wDwn9CIANuHlt',
+             'channel': 'stable',
+             'revision': '123'},
+            {'name': 'snapcraft',
+             'confinement': 'classic',
+             'id': '3lljuRvshPlM4gpJnH5xExo0DJBOvImu',
+             'channel': 'edge',
+             'revision': '345'},
+        ]
+
+        self.make_containerbuild().execute()
+
+        if hasattr(self, 'cross') and self.cross:
```

Comment:
What is this doing, exactly? What is this attribute?

Reply:
This is using the same approach as the other test cases in this file. The snaps are installed from the store and not from a local file in the case of a remote with another architecture.

Reply:
Oh, I didn't realize they're not used in the ProjectTestCase. Adding.